channeldb/payment_control.go

```diff
@@ -449,23 +442,22 @@ func (p *PaymentControl) updateHtlcKey(paymentHash lntypes.Hash,
 			return fmt.Errorf("htlcs bucket not found")
 		}
 
-		htlcBucket := htlcsBucket.NestedReadWriteBucket(htlcIDBytes)
-		if htlcBucket == nil {
+		if htlcsBucket.Get(append(htlcAttemptInfoKey, htlcIDBytes...)) == nil {
```

Comment:
I think this change would actually be bad for `bbolt`'s performance. This would mean all HTLCs would be in the same top-level bucket. This makes write operations very memory intensive with `bbolt` when there are many entries as it needs to copy its whole btree when adding new leaves.
But I can definitely see why this would help `etcd`. Not sure what other options we have. Thought if this is a performance bottleneck then it's definitely worth brainstorming a bit more.
Or perhaps even decide that `etcd` performance has precedence here if the performance hit in bbolt is not too dramatic.

Reply:
The variable names may be a bit misleading, but the htlcs are in the payment's own bucket (where the key is the payment hash). Before it was the same but there were three sub buckets per htlc attempt.